appcontext/context.go

```diff
@@ -64,3 +68,34 @@ func (context *Context) MarkAsPushed() error {
 		return errors.New("No EventID or EventChecksum Provided")
 	}
 }
+
+// PushToCoreData pushes the provided value as an event to CoreData using the device name and device resource name that have been set.
+func (context *Context) PushtoCoreData(deviceName string, readingName string, value interface{}) (string, error) {
+	now := time.Now().UnixNano()
+	val, err := util.CoerceType(value)
+	if err != nil {
+		return "", err
+	}
+	newReading := models.Reading{
+		Value:  string(val),
+		Origin: now,
+		Device: deviceName,
+		Name:   readingName,
+	}
+
+	readings := make([]models.Reading, 0, 1)
+	readings = append(readings, newReading)
+
+	newEdgeXEvent := &models.Event{
+		Device:   deviceName,
+		Origin:   now,
+		Readings: readings,
+	}
+
+	newEdgeXEvent.Readings = readings
+
+	correlation := uuid.New().String()
+	ctx := syscontext.WithValue(syscontext.Background(), clients.CorrelationHeader, correlation)
+	result, err := context.EventClient.Add(newEdgeXEvent, ctx)
+	return result, err
```

Comment:
Looks like the result from EventClient Add is just the ID assigned. Should we set it in our new Event and return the Event here so it is passed to the next function in the Pipeline when using PushToCore function?

Reply:
Yeah i like that idea

Reply:
Note: the new event wont have a correlationId, not sure if that matters...

Reply:
I thought that is what lines 97 & 98 are setting up?

Reply:
Ahh, need to also set it in the AppSDK Context...

Reply:
Yeah, but i'm not sure thats the right thing to do -- change the correlationId in the context

Reply:
Hummm, I think so since the original event has been replaced my the new event. Or use the old correlation id for the new event... NOT! ;-)

Reply:
what about the EventId on the context? EventChecksum? If I receive a CBOR event, and push a non cbor event to core? I dont think changing anything on the context is the right move, especially since i may not be done working on the original event that came in. All logs related to the pipeline execution would then reflect the new correlationId which would not be correct

Reply:
the context is supposed to reflect the event the triggered the pipeline

Reply:
So should the new event then be passed to the next function or pass the source data which goes to the current context?

Reply:
Also, i don't think we've "replaced" the event anymore than a transformtoJson or encrypt or compress function has "replaced" it. Its simply an action that has occurred